src/transformStateWithClones.js

```diff
@@ -1,13 +1,36 @@
 'use strict';
-
 /**
  * @param {Object} state
  * @param {Object[]} actions
- *
- * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const resultStates = [];
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = {
+          ...currentState, ...action.extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+
+      case 'clear':
+        currentState = {};
+        break;
+
+      default:
+        throw new Error('Unknown action');
```

Comment:
Usually it's a bad idea to throw without any context, because this means that consumer of this function (You in the different part of the code, or someone else in the team) has to know that this function is not safe and has to check if it will not throw.
I would change it to `console.error('Your message')`

Reply:
You can disable linting for this line only.